crates/blockifier/src/execution/native_syscall_handler.rs

```diff
@@ -35,28 +85,114 @@ impl<'state> StarkNetSyscallHandler for NativeSyscallHandler<'state> {
         todo!("Native syscall handler - deploy")
     }
 
-    fn replace_class(&mut self, _class_hash: Felt, _remaining_gas: &mut u128) -> cairo_native::starknet::SyscallResult<()> {
-        todo!("Native syscall handler - replace_class")
+    fn replace_class(
+        &mut self,
+        class_hash: Felt,
+        _remaining_gas: &mut u128,
+    ) -> cairo_native::starknet::SyscallResult<()> {
+        let class_hash = ClassHash(StarkHash::from(felt_to_starkfelt(class_hash)));
+        let contract_class = self
+            .state
+            .get_compiled_contract_class(class_hash)
+            .map_err(|_| vec![Felt::from_hex(FAILED_TO_GET_CONTRACT_CLASS).unwrap()])?;
+
+        match contract_class {
+            ContractClass::V0(_) => Err(vec![Felt::from_hex(FORBIDDEN_CLASS_REPLACEMENT).unwrap()]),
+            ContractClass::V1(_) | ContractClass::V1Sierra(_) => {
+                self.state
+                    .set_class_hash_at(self.storage_address, class_hash)
+                    .map_err(|_| vec![Felt::from_hex(FAILED_TO_SET_CLASS_HASH).unwrap()])?;
+
```

Comment:
The `replace_class` logic in the syscall module should be updated as well. Currently `V1Sierra` has a todo